server/schema/getchangedslavezoneswithtsig.sql

```diff
@@ -0,0 +1,24 @@
+CREATE OR REPLACE FUNCTION GetChangedSlaveZonesWithTSIG(
+	nameservername varchar,
+	out change_id bigint,
+	out change_name varchar,
+	out change_changetime int,
+	out change_tsigkeyname varchar
+) RETURNS SETOF record AS $$
+DECLARE r RECORD;
+BEGIN
+	FOR r IN	SELECT slavezone_change.id, zone, changetime, domainmetadata.tsigkey_name as tsigkeyname FROM slavezone_change 
+			INNER JOIN nameserver ON nameserver_id = nameserver.id
+			INNER JOIN slavezone ON zone = slavezone.name LEFT JOIN domainmetadata ON slavezone.id = domainmetadata.domain_id AND domainmetadata.kind = 'slave'
+			WHERE nameserver.name = nameservername AND status = 'PENDING' 
+			ORDER BY changetime ASC, slavezone_change.id ASC
```

Comment:
After calling DeleteSlaveZone, the zone is removed from the slave zone table and there is an entry in the 'slavezone_change' table. However, this function doesn't return the change. This is probably because of ```INNER JOIN slavezone ON zone = slavezone.name```, the slave zone doesn't exist.